File: linearSearch.cpp
#include <bits/stdc++.h>
using namespace std;

int linearSearch(vector<int> v, int x)
{
    for (int i = 0; i < v.size(); i++)
    {
        if (v[i] == x)
        {
            return i;
        }    
    }
    return -1;
}
int main()
{
    vector<int> v = {1, 2, 3, 4, 5};
    int x = 5;
    int ans=linearSearch(v, x);
    if(ans==-1){
        cout<<"Not found";
    } else{
        cout<<x<<" found at index: "<<ans;
    }
}

File: LinearSearch.java
public class LinearSearch {
    public static int linearSearch(int arr[], int x) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == x)
                return i;
        }
        return -1;
    }

    public static void main(String args[]) {
        int array[] = { 2, 4, 0, 1, 9 };
        int x = 1;

        int result = linearSearch(array, x);

        if (result == -1)
            System.out.print("Element not found");
        else
            System.out.print("Element found at index: " + result);
    }
}
